web/client/selectors/featuregrid.js

```diff
@@ -202,36 +202,87 @@ export const isEditingAllowedSelector = (state) => {
 
 export const isViewportFilterActive = state => get(state, 'featuregrid.viewportFilter', null);
 
 export const isFilterByViewportSupported = state => mapTypeSelector(state) !== MapLibraries.CESIUM;
 
+export const spatialFieldFilters = state => get(state, 'query.filterObj.spatialField');
+
 export const viewportFilter = createShallowSelectorCreator(isEqual)(
     isViewportFilterActive,
     mapBboxSelector,
     projectionSelector,
-    state => get(state, 'query.filterObj.spatialField'),
+    spatialFieldFilters,
     describeSelector,
     isFilterByViewportSupported,
     (viewportFilterIsActive, box, projection, spatialField = [], describeLayer, viewportFilterIsSupported) => {
         const attribute = findGeometryProperty(describeLayer)?.name;
         const existingFilter = spatialField?.operation ? [spatialField] : spatialField;
         return viewportFilterIsActive && viewportFilterIsSupported ? {
             spatialField: [
-                ...existingFilter,
+                // avoid restricted area filter dupplication
+                ...existingFilter.filter(f => !f.viewport && !f.restrictedArea),
                 {
                     geometry: {
                         ...bboxToFeatureGeometry(box.bounds),
                         projection
                     },
                     attribute: attribute,
                     method: "Rectangle",
-                    operation: "INTERSECTS"
+                    operation: "INTERSECTS",
+                    viewport: true
+                }
+            ]
+        } : {};
+    }
+);
+
+export const restrictedAreaFilter = createShallowSelectorCreator(isEqual)(
+    restrictedAreaSelector,
+    spatialFieldFilters,
+    viewportFilter,
+    projectionSelector,
+    describeSelector,
+    state => restrictedAreaOperatorSelector(state),
+    (restrictedArea, spatialField = [], viewPortFilter, projection, describeLayer, operator) => {
+        const attribute = findGeometryProperty(describeLayer)?.name;
+        let existingFilter = [];
+        // if activate, viewportFilter already get existing filter
+        if (isEmpty(viewPortFilter) && !isEmpty(spatialField)) {
+            existingFilter = spatialField?.operation ? [spatialField] : spatialField;
+        }
+        return !isEmpty(restrictedArea) ? {
+            spatialField: [
+                ...existingFilter,
+                {
+                    geometry: {
+                        ...restrictedArea,
+                        projection: "EPSG:4326"
+                    },
+                    attribute: attribute,
+                    method: "Polygon",
+                    operation: operator || "CONTAINS",
+                    restrictedArea: true
                 }
             ]
         } : {};
     }
 );
+
+/**
+ * Create spatialField filters array.
+ * Contains filters from viewportFilter, restrictedArea, exsting WFS filter
+ */
+export const additionnalGridFilters = (state) => {
```

Comment:
Using this approach makes the sync with map not working. While it still makes sense with viewport filter (because the map already shows the features in map), here this makes things a little unconsistent, and it could be practical for the user to see only the allowed rows on the map. What do you think?

A more functional approach is to add / remove the filter to the `query` object, as well as the other filters, like quick filter, do. 
![image](https://github.com/user-attachments/assets/1e46db98-300b-47b3-a22d-35fee67102e0)

The  `filters` [section](https://docs.mapstore.geosolutionsgroup.com/en/v2024.02.00/developer-guide/LayerFilter/)  recently added to the filter may help you to manage it in a easier way.

Basically you can add to the filter a section filters: that can contain and use cql filter to apply some additional logic. 
You can identify these filters by id, adding your own property, to manage them (e.g. adding/removing).

Reply:
> it could be practical for the user to see only the allowed rows on the map. What do you think?

Exactly, this is the expected behavior

> Using this approach makes the sync with map not working. 

Not really sur to understand. additionnalGridFilters methods is not dependent on the screen filter but but also uses it if activate. Maybe my code is wrong with I wanted to do.

> Basically you can add to the filter a section filters: that can contain and use cql filter to apply some additional logic.

I will change the code to use filters section as you expect.

Thanks for this proposals @offtherailz .